cmd/influxd/upgrade/upgrade.go

```diff
@@ -361,8 +361,9 @@ func runUpgradeE(cmd *cobra.Command, options *options, verbose bool) error {
 		}
 	}
 
-	var v1Config *configV1
+	v1Config := &configV1{}
```

Comment:
This (making sure `v1Config` isn't `nil`) is the fix for the panic

Reply:
The panic was occurring on [this line](https://github.com/influxdata/influxdb/pull/20548/files#diff-e5c52e2be1a2771a344e5d0402468658d34a70fe9ba50453edb3b3b3e97a4eaeR476) (`nil` dereference on `!v1Config.Http.AuthEnabled`)